src/reporters/apiary-reporter.coffee

```diff
@@ -222,11 +248,28 @@ class ApiaryReporter
       if @verbose
         logger.log 'Starting REST Reporter HTTPS Request'
       req = https.request options, handleRequest
+
+      req.on 'error', (error) =>
+        @serverError = true
+        if error.code == 'ECONNREFUSED'
+          logger.error "Apiary reporter: Error connecting to Apiary test reporting API."
+          callback()
+        else
+          return callback error, req, null
+
     else
       if @verbose
         logger.log 'Starting REST Reporter HTTP Response'
       req = http.request options, handleRequest
 
+      req.on 'error', (error) =>
+        @serverError = true
+        if error.code == 'ECONNREFUSED'
+          logger.error "Apiary reporter: Error connecting to Apiary test reporting API."
+          callback()
+        else
+          callback error, req, null
+
```

Comment:
@netmilk Can you please write it more DRY-way? The `HTTP` vs `HTTPS` share a lot of logic, but differ only in the message written to verbose `logger.log`. Thanks..